### ISSUE TITLE
DG1 compatibility?

### ISSUE THREAD

**Issue by mo-g** [intel/intel-extension-for-tensorflow#18]
Is the Intel DG1 hardware (notebook, desktop, and the related HPC cards) compatible either functionally, or even hypothetically in future?

I'm basically looking at <$100, <40W hardware - GT1030, P620, DG1.

**Comment by mo-g** [comment]
@yiqianglee Haven't tried, I don't own one yet!

I don't have a 10G or newer platform, so the card would be useless to me for anything other than compute workloads, didn't want to throw away $100 if it wasn't gonna work!

**Comment by guizili0** [comment]
@mo-g I have checked on DG1 with our 1.0 release.
Functionality can work. 
For performance, the first is very slow due to we did not enable DG1 AOT in our release whl, you can try to build a whl with DG1 AOT to improve the performance here. You can refer our doc here https://intel.github.io/intel-extension-for-tensorflow/latest/get_started.html

I try this TF example: https://www.tensorflow.org/tutorials/images/classification
the result is:
92/92 [==============================] - 171s 342ms/step - loss: 1.3340 - accuracy: 0.4448 - val_loss: 1.1196 - val_accuracy: 0.5518
Epoch 2/10
92/92 [==============================] - 15s 165ms/step - loss: 0.9764 - accuracy: 0.6110 - val_loss: 0.9627 - val_accuracy: 0.5995
Epoch 3/10
92/92 [==============================] - 15s 166ms/step - loss: 0.7529 - accuracy: 0.7187 - val_loss: 0.8497 - val_accuracy: 0.6798
Epoch 4/10
92/92 [==============================] - 15s 166ms/step - loss: 0.4925 - accuracy: 0.8263 - val_loss: 1.0424 - val_accuracy: 0.6553
Epoch 5/10
92/92 [==============================] - 15s 166ms/step - loss: 0.3003 - accuracy: 0.8961 - val_loss: 1.0376 - val_accuracy: 0.6662
Epoch 6/10
92/92 [==============================] - 15s 165ms/step - loss: 0.1662 - accuracy: 0.9496 - val_loss: 1.1430 - val_accuracy: 0.6580
Epoch 7/10
92/92 [==============================] - 15s 166ms/step - loss: 0.1250 - accuracy: 0.9612 - val_loss: 1.5662 - val_accuracy: 0.6458
Epoch 8/10
92/92 [==============================] - 15s 165ms/step - loss: 0.0442 - accuracy: 0.9884 - val_loss: 1.5190 - val_accuracy: 0.6294
Epoch 9/10
92/92 [==============================] - 15s 165ms/step - loss: 0.0177 - accuracy: 0.9969 - val_loss: 1.8557 - val_accuracy: 0.6281
Epoch 10/10
92/92 [==============================] - 15s 164ms/step - loss: 0.0128 - accuracy: 0.9980 - val_loss: 1.8669 - val_accuracy: 0.6485

**Comment by mo-g** [comment]
Thanks for this, @guizili0. I'll take a look. One last question on this - [Legacy releases are for packages that are verified for Intel Gen9/DG1 graphics.Packages are published to the legacy repository when milestones are achieved.](https://dgpu-docs.intel.com/releases_legacy/index.html)

I take it based on that that I shouldn't expect this kind of thing to work long-term?

**Comment by guizili0** [comment]
@mo-g Sorry for the late update. From the official driver, DG1 driver has been moved to stable part.
https://dgpu-docs.intel.com/releases/stable_540_20221205.html

**Comment by mo-g** [comment]
Thanks very much @guizili0! I'll close the ticket now, and put a DG1 on my shopping list for the compute cluster. :)
